core/src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -63,7 +76,81 @@ public ClientBuilder(String uri) {
+        this(Scheme.of(format, protocol), requireNonNull(endpoint, "endpoint"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the HTTP client that connects to the specified
+     * {@link Endpoint} with the {@link SessionProtocol}.
+     */
+    public ClientBuilder(SessionProtocol protocol, Endpoint endpoint) {
+        this(requireNonNull(protocol, "protocol"), SerializationFormat.NONE,
+             requireNonNull(endpoint, "endpoint"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@code Scheme}.
+     */
+    public ClientBuilder(String scheme, Endpoint endpoint) {
+        this(Scheme.parse(scheme), requireNonNull(endpoint, "endpoint"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@link Scheme}.
+     */
+    public ClientBuilder(Scheme scheme, Endpoint endpoint) {
+        this(null, requireNonNull(scheme, "scheme"), requireNonNull(endpoint, "endpoint"), "");
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@link SessionProtocol}, {@link SerializationFormat}, and {@code path}.
+     */
+    public ClientBuilder(SessionProtocol protocol, SerializationFormat format, Endpoint endpoint, String path) {
+        this(Scheme.of(format, protocol), requireNonNull(endpoint, "endpoint"), requireNonNull(path, "path"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the HTTP client that connects to the specified
+     * {@link Endpoint} with the {@link SessionProtocol} and {@code path}.
+     */
+    public ClientBuilder(SessionProtocol protocol, Endpoint endpoint, String path) {
+        this(requireNonNull(protocol, "protocol"), SerializationFormat.NONE,
+             requireNonNull(endpoint, "endpoint"), requireNonNull(path, "path"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@code scheme} and {@code path}.
+     */
+    public ClientBuilder(String scheme, Endpoint endpoint, String path) {
+        this(Scheme.parse(scheme), requireNonNull(endpoint, "endpoint"), requireNonNull(path, "path"));
+    }
+
+    /**
+     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
+     * {@link Endpoint} with the {@link Scheme} and {@code path}.
+     */
+    public ClientBuilder(Scheme scheme, Endpoint endpoint, String path) {
+        this(null, requireNonNull(scheme, "scheme"), requireNonNull(endpoint, "endpoint"),
+             requireNonNull(path, "path"));
+    }
+
+    private ClientBuilder(URI uri, Scheme scheme, Endpoint endpoint, String path) {
```

Comment:
- You need to annotate the parameters with `@Nullable`.
- Could you add an assertion like:

  ```java
  assert uri != null || scheme != null && endpoint != null && path != null;
  ```